scarb/src/manifest_editor/add.rs

```diff
@@ -128,6 +130,17 @@ impl Dep {
                 DefaultBranch
             };
 
+            let git = match CanonicalUrl::new(&Url::parse(&git)?) {
```

Comment:
Failures to parse the URL will cause this operation to error, while I think we would still rather print just a warning here.

Reply:
Hmm.. was considering that but is there a reason not to panic if it cannot be parsed as url at all?

Reply:
My reasoning here is that user might make a typo and I think I wouldn't yell at his face for this at this stage? Rather print some warnings and let him peacefully fix a mistake in the edited file.

Reply:
I'm open to being convinced that I think wrong 🙃

Reply:
i feel like simply giving just warning would make user assumes that its not that big of a problem... and also maybe somewhat counter intuitive to still write to the manifest when we know it will eventually panic? wdyt?

but this one also has same the problem as above... will eventually panic when resolving workspace

Reply:
Scarb already follows a pattern that is prints warnings if it knows that future steps will likely fail. For example if you use non-lib dependency. 

I'm not sure. Generally I'd like the editor to avoid failures as much as possible and only guide what might go wrong

Reply:
I see... I didnt realize about that pattern actually.